graph-builder/tests/net/quay_io/mod.rs

```diff
@@ -73,6 +76,16 @@ fn expected_releases(
     releases
 }
 
+fn remove_metadata_by_key(releases: &mut Vec<Release>, key: &str) -> Vec<Release> {
```

Comment:
I'm not sure I follow the idea behind this, but I think this will result in a new cloned vector. If that is the case, I think this would better take ownership of the input vector and return it back after manipulation. Or otherwise just mutate the input one in-place.